src/icalendar/attr.py

```diff
@@ -1,24 +1,308 @@
+
+    ``start`` can be a :class:`datetime.date` or a :class:`datetime.datetime`,
+    with and without timezone.
+
+    ``end`` is :obj:`None` if the end is not specified and a :class:`datetime.datetime`
+    if the end is specified.
+
+    Value Type:
+        The default value type for this property is DATE-TIME.
+        The value type can be set to DATE or PERIOD.
+
+    Property Parameters:
+        IANA, non-standard, value data type, and time
+        zone identifier property parameters can be specified on this
+        property.
+
+    Conformance:
+        This property can be specified in recurring "VEVENT",
+        "VTODO", and "VJOURNAL" calendar components as well as in the
+        "STANDARD" and "DAYLIGHT" sub-components of the "VTIMEZONE"
+        calendar component.
+
+    Description:
+        This property can appear along with the "RRULE"
+        property to define an aggregate set of repeating occurrences.
+        When they both appear in a recurring component, the recurrence
+        instances are defined by the union of occurrences defined by both
+        the "RDATE" and "RRULE".
+
+        The recurrence dates, if specified, are used in computing the
+        recurrence set.  The recurrence set is the complete set of
+        recurrence instances for a calendar component.  The recurrence set
+        is generated by considering the initial "DTSTART" property along
+        with the "RRULE", "RDATE", and "EXDATE" properties contained
+        within the recurring component.  The "DTSTART" property defines
+        the first instance in the recurrence set.  The "DTSTART" property
+        value SHOULD match the pattern of the recurrence rule, if
+        specified.  The recurrence set generated with a "DTSTART" property
+        value that doesn't match the pattern of the rule is undefined.
+        The final recurrence set is generated by gathering all of the
+        start DATE-TIME values generated by any of the specified "RRULE"
+        and "RDATE" properties, and then excluding any start DATE-TIME
+        values specified by "EXDATE" properties.  This implies that start
+        DATE-TIME values specified by "EXDATE" properties take precedence
+        over those specified by inclusion properties (i.e., "RDATE" and
+        "RRULE").  Where duplicate instances are generated by the "RRULE"
+        and "RDATE" properties, only one recurrence is considered.
+        Duplicate instances are ignored.
+
+    .. note::
+
+        You cannot modify the RDATE value by modifying the result.
+        Use :func:`icalendar.Component.add` to add values.
+
+    Example::
+
+        >>> from icalendar import Event
+        >>> from datetime import datetime
+        >>> event = Event()
+
+        # Add a list of recurrence dates
+        >>> event.add("RDATE", [datetime(2025, 4, 28, 16, 5)])
+        >>> event.rdates
+        [(datetime.datetime(2025, 4, 28, 16, 5), None)]
```

Comment:
Indent for consistency with Description and use explicit lexer.

```suggestion
    Example:

        .. code-block:: pycon

            >>> from icalendar import Event
            >>> from datetime import datetime
            >>> event = Event()

            # Add a list of recurrence dates
            >>> event.add("RDATE", [datetime(2025, 4, 28, 16, 5)])
            >>> event.rdates
            [(datetime.datetime(2025, 4, 28, 16, 5), None)]
```